scripts/mods/mod_dynamic_perks/classes/perk_tree.nut

```diff
@@ -232,6 +232,17 @@ this.perk_tree <- ::inherit(::MSU.BBClass.Empty, {
 		return ret;
 	}
 
+	function getPerkGroupID(_perkID)
```

Comment:
A perk can belong to multiple perk groups. So it would be good to highlight all of them (maybe with different colors). So this should return an array.

Alternatively, at the start of the game `AfterHooks`, we populate every `perkDef` with an array `PerkGroups` which contains the ids of the perk groups it belongs to. Then this entire process becomes simpler as the perk groups of a perk are always present in its perk def.

Reply:
This shows the perk group on this bro that the perk belongs to, much like the 'from the xx group' tooltip.

Reply:
- The `From the xx group` tooltip mentions **all** perk groups in the game that this perk belongs to.
- Even on a single bro a perk can be on multiple groups. E.g. the Phalanx perk is present on both Militia and Shield perk groups and the Militia background has both these perk groups.

Reply:
Okay but that turns the feature from a very neat thing I can implement in one go to this annoying mess where I need to look up each perk and each of its perk groups to regroup them into matching subgroups and how to handle two different per geoups do I now need two different color codes and argh

Reply:
Yeah that definitely makes it more complicated. And the cases where that happens is not that often tbh (i.e. where a bro has a perk belonging to multiple groups). But would be nice to have handling for that.
- I'd say we do what I suggested above i.e. after hooks we go through all perk groups, and add to each perkDef an array called `PerkGroups` which lists all the ids of the perk groups where this perk can be found. Then during this highlighting phase, you just iterate over all the perk groups this perk has one by one and highlight all the other perks which belong to that perk group. Then switch color and iterate the second perk group.
- For the color I'd say just turn the hue by a few degrees for every subsequent perk group you highlight.

Reply:
Ok sure whatever I implemented it, just give me a bunch of colors, preferably in rgba.
Hue doesn't cut it, some people don't see well.

Reply:
- 255,234,125,0.5
- 255,125,125,0.5
- 125,125,255,0.5
- 125,255,125,0.5

That should cover up to 4 perk groups which is more than enough imo.

Reply:
Yeah something was wrong with those so I just went with generic colors.

Reply:
It didn't end up being that bad to implement, but certainly messier than before. I disabled highlighting the initial perk also, though might want to turn it on if there's only a single perk group

Reply:
Thanks, I've pushed a commit with some minor adjustments